File: Base Classes/MainActivity.java
package com.example.myapplication;

public class MainActivity extends BaseActivity {

    private static final String TAG = "MainActivity";
    private static final String TEXT = "Hello World!";

    @Override
    protected int getLayoutResID() {
        return R.layout.activity_main;
    }

    @Override
    protected void initViews() {
        /*
         *  UI created
         *  your code
         */
        binding.tvText.setText(TEXT);
    }
}

File: Base Classes/MainActivity.kt
package com.example.myapplication

class MainActivity : BaseActivity(R.layout.activity_main) {

    companion object {
        const val TAG = "MainActivity"
        const val TEXT = "Hello World!"
    }
    
    override fun initViews() {
        /*
         *  UI created
         *  your code
         */
        binding?.tvText?.text = TEXT
    }
}
